core/store/models/job_run.go

```diff
@@ -67,8 +64,8 @@ func (jr JobRun) ForLogger(kvs ...interface{}) []interface{} {
 		output = append(output, "observed_height", jr.ObservedHeight.ToInt())
 	}
 
-	if jr.Result.HasError() {
-		output = append(output, "job_error", jr.Result.Error())
+	if jr.Result.ErrorMessage.Valid {
+		output = append(output, "job_error", jr.Result.ErrorString())
```

Comment:
Maybe an HasError() would be clearer? or something like 'err, ok := Result.GetError()'

Reply:
Seconded.